staging/src/k8s.io/kubectl/pkg/kuberc/marshal.go

```
@@ -86,6 +91,13 @@ func decodePreference(kubercFile string) (*config.Preference, error) {
 			continue
 		}
 
+		// if there was a strict decode error, check check whether the
+		// client-go credential plugin allowlist is the cause; if so, the
+		// allowlist is misconfigured and the operation should fail.
```

Should fail _why_?  Because unlike the rest of the preferences, this a security control.  Users don't like their security controls to be silently ignored.

---

Good call, thank you.

---

done